tutorials/3d/spring_arm.rst

```diff
@@ -0,0 +1,131 @@
+
+Every physics process frame, the spring arm will perform a motion cast to check if anything is collided with:
+
+.. image:: img/spring_arm_camera_motion_cast.webp
+
+When the shape hits something, the camera will be placed at or near the collision point:
+
+.. image:: img/spring_arm_camera_collision.webp
+
+Setting up the spring arm and camera
+------------------------------------
+
+Let's add a spring arm camera setup to the platformer demo.
+
+.. note:: You can download the Platformer 3D demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
+
+In general, for a third-person camera setup, you will have three nodes as children of the node that you're following:
+
+- `Node3D` (the "pivot point" for the camera)
+
+    - `SpringArm3D`
+
+        - `Camera3D`
+
+Open the ``player/player.tscn`` scene. Set these up as children of our player and give them unique names so we can find them in our script. **Make sure to delete the existing camera node!**
+
+.. image:: img/spring_arm_editor_setup.webp
+
+Let's move the pivot point up by ``3`` on the Y-axis so that it's not on the ground:
+
+.. image:: img/spring_arm_pivot_setup.webp
+
+
+Give the spring arm a length of ``3`` so that it is placed behind the character:
+
+.. image:: img/spring_arm_length_setup.webp
+
+.. note:: Leave the **Shape** of the spring arm as ``<empty>``. This way, it will use the camera's pyramid shape.
+
+    If you want, you can also try other shapes - a sphere is a common choice since it slides smoothly along edges.
+
+Update the top of ``player/player.tscn`` to grab the camera and the pivot points by their unique names:
+
+.. code-block:: gdscript
+    :caption: player/player.tscn
+
+    # remove this existing camera line
+    # @onready var _camera := $Target/Camera3D as Camera3D
+
+    @onready var _camera := %Camera3D as Camera3D
+    @onready var _camera_pivot := %CameraPivot as Node3D
+
+Add an ``_unhandled_input`` function to check for camera movement and then rotate the pivot point accordingly:
+
+.. code-block:: gdscript
+    :caption: player/player.tscn
+
+    @export_group("Camera")
+    @export_range(0.0, 1.0) var mouse_sensitivity := 0.01
+    @export var tilt_upper_limit := PI / 3.0
+    @export var tilt_lower_limit := -PI / 3.0
+
+
+    func _unhandled_input(event: InputEvent) -> void:
+        if event is InputEventMouseMotion:
+            _camera_pivot.rotation.x -= event.relative.y * mouse_sensitivity
+
+            # this will keep the camera from rotating too far up or down
+            _camera_pivot.rotation.x = clampf(
+                _camera_pivot.rotation.x, tilt_lower_limit, tilt_upper_limit
+            )
+
+            _camera_pivot.rotation.y += -event.relative.x * mouse_sensitivity
+
```

Comment:
```suggestion
.. code-block:: gdscript
    :caption: player/player.gd

@export_range(0.0, 1.0) var mouse_sensitivity = 0.01
@export var tilt_limit = deg_to_rad(75)

func _unhandled_input(event: InputEvent) -> void:
	if event is InputEventMouseMotion:
		_camera_pivot.rotation.x -= event.relative.y * mouse_sensitivity
		# Prevent the camera from rotating too far up or down.
		_camera_pivot.rotation.x = clampf(_camera_pivot.rotation.x, -tilt_limit, tilt_limit)
		_camera_pivot.rotation.y += -event.relative.x * mouse_sensitivity
```
Some code style nitpicks:

- We're actually editing the `.gd`, not the scene
- Use dynamic typing. I don't love it either, but it's the [current guideline](https://docs.godotengine.org/en/stable/contributing/documentation/docs_writing_guidelines.html#dynamic-vs-static-typing)
- Export group is a detail that doesn't matter for this example (even if it would be good practice in a larger script)
- Use a single tilt limit
- IMO deg_to_rad on a round multiple of 5 looks cleaner than a multiple of PI. This one's subjective, feel free to ignore
- Comments are sentence case.
- Make the one-liner actually a one-liner